src/traveler.js

```diff
@@ -1,49 +1,106 @@
+
 class Traveler extends User {
-  constructor(userInfo, tripHistory) {
-    // console.log('I am in user=>',tripHistory)
+  constructor(userInfo, tripHistory, destinationsData) {
+    // console.log(destinationsData)//object
     super(userInfo);
-    this.id = userInfo.id;
-    this.name = userInfo.name;
+    // this.id = userInfo.id;
+    // this.name = userInfo.name;
     this.travelerType = userInfo.travelerType;
+    this.today = moment().format('YYYY-MM-DD');
     this.travelHistory = tripHistory.userTripHistory || [];
-    this.tripsHistory = []; //past trips
-    this.currentTrip  = {} ; // || [] probably it will be just an object
-    this.upcomingTrips = [];
-    this.pendingTrips = [];
+    this.destinationsData = destinationsData;
+    this.tripsHistory = this. getPastTrips(); //past trips
+    this.currentTrip  = this.getCurrentTrip() || []; // || [] probably it will be just an object
+    this.upcomingTrips = this.getFutureTrips() || [];
+    // this.pendingTrips =  [];// i might not need this since the method is dynamic enough to get pending and aporved trips
+    this.tripsThisYear = this.getTripByYears() || []; //for 2020
     this.spentOverYear = 0;
-    // console.log(this.travelHistory)
   }
    //what format makes more sense? //myabe having and array of 
-   /*[ "2020/10/04":{data},  "2020/10/04":{data},  "2020/10/04":{data}, date:{data}]*/
+   /*data : { "2020/10/04":{data},  "2020/10/04":{data},  "2020/10/04":{data}, date:{data}}*/
 
   restructuredTripHistoryByDate() {
-    return this.travelHistory.reduce((newData, entry) => {
-      newData.push({[entry.date] : entry})
+    const data = this.travelHistory.reduce((newData, entry) => {
+      newData[entry.date] = entry
       return newData
-    }, [])
-  }
+    }, {})
 
+    return data
+  }
   //get currentTrip
+  //currently not working as intended
   getCurrentTrip() {
-    
+   const currentTrip = this.travelHistory.reduce((theTrip, trip) => {
+    let startDate = moment(trip.date, 'YYYY-MM-DD').format('YYYY-MM-DD');
+    let endDate = moment(startDate, 'YYYY-MM-DD').add(trip.duration, 'days').format('YYYY-MM-DD');
+    // if (moment(this.today, 'YYYY-MM-DD').isAfter(startDate, 'YYYY-MM-DD') && moment(this.today, 'YYYY-MM-DD').isBefore(endDate, 'YYYY-MM-DD')) {
+    //   theTrip.push(trip)
+    // }
+    //this 👇🏽 is the same as th if statement above 🤦🏽‍♂️
+    if (moment(this.today, 'YYYY-MM-DD').isBetween(startDate, endDate)) {
+      theTrip.push(trip)
+    }
+    return theTrip
+   },[])
+    return currentTrip
+  }
+
+   //get futureTrips
+   getFutureTrips() {
+     const trips = this.travelHistory.reduce((tripsList, trip) => {
+      //  let startDate = moment(trip.date, 'YYYY-MM-DD').format('YYYY-MM-DD');
+       if (moment(trip.date, 'YYYY-MM-DD').isAfter(this.today, 'YYYY-MM-DD')) {
+         tripsList.push(trip)
+       }
+       return tripsList
+      },[])
+
+       return trips
+     }
+   
+  //get pastTrips
+  getPastTrips(){
```

Comment:
I don't think I've seen .isBefore before. This is a very good use of moment in this application.

Reply:
I remember learning about it during the `activity tracker` project and I talked to my mentor and suggested me to use it for this project. It definitely made the code more readable for me, however, I ran into some problems getting it to work but figured it out at the end. The problem I had was that the data had to be in a specific format `(YYYY-MM-DD)` and not `(YYYY/MM/DD)`. It was really easy after figuring that part out 👍